ui/components/src/Table/TableHeaderCell.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { TableSortLabel, Typography, tableSortLabelClasses } from '@mui/material';
+import { SortDirection } from './model/table-model';
+import { TableCell, TableCellProps } from './TableCell';
+
+export interface TableHeaderCellProps extends TableCellProps {
+  canSort?: boolean;
```

Comment:
Can we just check if `onSort` is defined or not to see if it should be sortable?

Reply:
Good call! I'll refactor that bit.